src/result.rs

```diff
@@ -69,72 +77,97 @@ pub enum CqlType {
 
 #[napi]
 impl QueryResultWrapper {
-    pub fn from_query(internal: QueryResult) -> QueryResultWrapper {
-        QueryResultWrapper { internal }
+    pub fn from_query(internal: QueryResult) -> napi::Result<QueryResultWrapper> {
+        let value = match internal.result_not_rows() {
+            Ok(_) => QueryResultVariant::EmptyResult(internal),
+            // We can unwrap, as error will be returned only if result_not_rows is OK
+            Err(_) => {
+                QueryResultVariant::RowsResult(internal.into_rows_result().map_err(err_to_napi)?)
+            }
+        };
+        Ok(QueryResultWrapper { internal: value })
```

Comment:
:wrench: This is logically correct, but not rusty (no clear exhaustive pattern matching). What should be done here is pattern matching on `internal.into_rows_result()`.

Reply:
Then we can't avoid cloning, as matching on it consumes it, and we can't create empty result from it

Reply:
[Have you looked at the error variants that can arise there?](https://docs.rs/scylla/latest/scylla/transport/query_result/enum.IntoRowsResultError.html)
`IntoRowsResultError::ResultNotRows(QueryResult)` allows for getting the `QueryResult` back.